### PR TITLE
[WIP] Fix layout calculation.

### PR DESCRIPTION
**Pull request by bobocandys**
Make cellWidth under layout to be fixed when we automatically augment domain.

Test: All time units but date, date currently is not doing something unexpected.

Issue: #981
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/vega/vega-lite/pull/1058%23issuecomment-189579201%22%2C%20%22https%3A//github.com/vega/vega-lite/pull/1058%23issuecomment-189585374%22%2C%20%22https%3A//github.com/vega/vega-lite/pull/1058%23issuecomment-189732884%22%2C%20%22https%3A//github.com/vega/vega-lite/pull/1058%23issuecomment-210672158%22%2C%20%22https%3A//github.com/vega/vega-lite/pull/1058%23discussion_r51683455%22%2C%20%22https%3A//github.com/vega/vega-lite/pull/1058%23discussion_r52821271%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/vega/vega-lite/pull/1058%23issuecomment-189579201%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Any%20progress%20on%20this%3F%20%22%2C%20%22created_at%22%3A%20%222016-02-27T05%3A26%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/589034%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/domoritz%22%7D%7D%2C%20%7B%22body%22%3A%20%22Sorry%20not%20yet.%20I%20have%20been%20busy%20on%20my%20course%20work.%5CnOn%20Fri%2C%20Feb%2026%2C%202016%20at%2021%3A26%20Dominik%20Moritz%20%3Cnotifications%40github.com%3E%5Cnwrote%3A%5Cn%5Cn%3E%20Any%20progress%20on%20this%3F%5Cn%3E%5Cn%3E%20%5Cu2014%5Cn%3E%20Reply%20to%20this%20email%20directly%20or%20view%20it%20on%20GitHub%5Cn%3E%20%3Chttps%3A//github.com/vega/vega-lite/pull/1058%23issuecomment-189579201%3E.%5Cn%3E%5Cn%22%2C%20%22created_at%22%3A%20%222016-02-27T06%3A30%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7077283%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/bobocandys%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40domoritz%20%20%20Actually%20I%20have%20already%20fixed%20this%20but%20haven%27t%20copy%20her%20test%20case%20into%20the%20master%2C%20so%20haven%27t%20close%20this.%20%20%20But%20layout%20would%20be%20vastly%20refactored%20for%20composition%20anyway%20so%20let%27s%20wait%20on%20that.%20%20%22%2C%20%22created_at%22%3A%20%222016-02-27T21%3A39%3A44Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/111269%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kanitw%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40kanitw%20Can%20we%20copy%20this%20over%20and%20close%20this%20pr%3F%22%2C%20%22created_at%22%3A%20%222016-04-15T22%3A41%3A01Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/589034%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/domoritz%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2030f4f750641a943001541f390547cda78233f24f%20src/compile/data.ts%204%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/vega/vega-lite/pull/1058%23discussion_r52821271%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Hmm%2C%20what%20about%20y%3F%20%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222016-02-13T02%3A22%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/111269%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kanitw%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/compile/data.ts%3AL252-263%22%7D%2C%20%22Pull%20f9c09af76db1d83b9a931077c7053190991bb485%20test/compile/data.test.ts%2014%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/vega/vega-lite/pull/1058%23discussion_r51683455%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%27Handle%20fixed%20time%20unit%20domain.%27%20%20%3A%29%20%22%2C%20%22created_at%22%3A%20%222016-02-03T06%3A34%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/111269%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kanitw%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/compile/data.test.ts%3AL348-558%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull f9c09af76db1d83b9a931077c7053190991bb485 test/compile/data.test.ts 14'></a> &nbsp;<b><a href='https://github.com/vega/vega-lite/pull/1058#discussion_r51683455'>File: test/compile/data.test.ts:L348-558</a></b>
- <a href='https://github.com/kanitw'></a> 'Handle fixed time unit domain.'  :)
- [ ] <a href='#crh-comment-Pull 30f4f750641a943001541f390547cda78233f24f src/compile/data.ts 4'></a> &nbsp;<b><a href='https://github.com/vega/vega-lite/pull/1058#discussion_r52821271'>File: src/compile/data.ts:L252-263</a></b>
- <a href='https://github.com/kanitw'></a> Hmm, what about y?
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/vega/vega-lite/pull/1058#issuecomment-189579201'>General Comment</a></b>
- <a href='https://github.com/domoritz'></a> Any progress on this?
- <a href='https://github.com/bobocandys'></a> Sorry not yet. I have been busy on my course work.
On Fri, Feb 26, 2016 at 21:26 Dominik Moritz <notifications@github.com>
wrote:
> Any progress on this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vega/vega-lite/pull/1058#issuecomment-189579201>.
>
- <a href='https://github.com/kanitw'></a> @domoritz   Actually I have already fixed this but haven't copy her test case into the master, so haven't close this.   But layout would be vastly refactored for composition anyway so let's wait on that.
- <a href='https://github.com/domoritz'></a> @kanitw Can we copy this over and close this pr?

<a href='https://www.codereviewhub.com/vega/vega-lite/pull/1058?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/vega/vega-lite/pull/1058?approve=1'></a>&nbsp;<a href='https://github.com/vega/vega-lite/pull/1058'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: test/compile/data.test.ts [hunk 1]
```diff
@@ -348,3 +348,210 @@ describe('data.summary', function () {
     });
   });
 });
+
+describe('Handle fix mode', function() {
```
Comment (kanitw) [inline test/compile/data.test.ts]: 'Handle fixed time unit domain.'  :)

File: src/compile/data.ts [hunk 2]
```diff
@@ -252,8 +252,11 @@ export namespace layout {
           ops: ['distinct']
         });
       }
+      const xTimeUnit = model.fieldDef(X).timeUnit;
```
Comment (kanitw) [inline src/compile/data.ts]: Hmm, what about y?

**Comment by domoritz** [comment]
Any progress on this?

**Comment by bobocandys** [comment]
Sorry not yet. I have been busy on my course work.

**Comment by kanitw** [comment]
@domoritz   Actually I have already fixed this but haven't copy her test case into the master, so haven't close this.   But layout would be vastly refactored for composition anyway so let's wait on that.

**Comment by domoritz** [comment]
@kanitw Can we copy this over and close this pr?

**Comment by kanitw** [comment]
I believe I have copied similar test.

**Pull request closed without merging**